pkg/user/repository.go

```diff
@@ -75,3 +75,14 @@ func (r repository) findById(id uint) (*model.User, error) {
 	}
 	return u, err
 }
+
+func (r repository) delete(id uint) error {
+	db := r.db.Unscoped().Delete(&model.User{}, id)
+	if db.Error != nil {
+		return db.Error
```

Comment:
Same as in https://github.com/dhis2-sre/im-manager/pull/253#discussion_r1221514101 applies. To me the rule is if the error does not originate from us (errors.New, fmt.Errorf) then we need to turn it into ours by calling `fmt.Errorf` or `errdef`. This way errors will be understandable for our users 😄